src/ScrollToCell.tsx

```diff
@@ -0,0 +1,54 @@
+import { useRef } from 'react';
+
+import { useLayoutEffect } from './hooks';
+import { scrollIntoView } from './utils';
+
+export interface PartialPosition {
+  readonly idx?: number | undefined;
+  readonly rowIdx?: number | undefined;
+}
+
+export default function ScrollToCell({
+  scrollToPosition: { idx, rowIdx },
+  gridElement,
+  setScrollToCellPosition
+}: {
+  scrollToPosition: PartialPosition;
+  gridElement: HTMLDivElement;
+  setScrollToCellPosition: (cell: null) => void;
+}) {
+  const ref = useRef<HTMLDivElement>(null);
+
+  useLayoutEffect(() => {
+    function removeScrollToCell() {
+      setScrollToCellPosition(null);
+    }
+
+    const observer = new IntersectionObserver(removeScrollToCell, {
+      root: gridElement,
+      threshold: 1.0
+    });
+
+    observer.observe(ref.current!);
+
+    return () => {
+      observer.disconnect();
+    };
+  }, [gridElement, setScrollToCellPosition]);
+
+  useLayoutEffect(() => {
```

Comment:
Should this effect run first?

Reply:
I was thinking the same. I tried both and it did not make a difference. I can move it